### PR TITLE
More mech refactoring

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

The ride never ends
Fixes hack for phasing by making it use the proper code
Removes a bunch of extraneous junk
Moves mech related honkers to mech code
more mech junk

## Changelog
:cl:
refactor: Changed how mechs handle movement and other cleanup
/:cl:

### CODE REVIEW

File: code/modules/vehicles/mecha/_mecha.dm [hunk 1]
```diff
@@ -77,7 +78,7 @@
 	///The setting of the valve on the internal tank
 	var/internal_tank_valve = ONE_ATMOSPHERE
 	///The internal air tank obj of the mech
-	var/obj/machinery/portable_atmospherics/canister/internal_tank
+	var/obj/machinery/portable_atmospherics/canister/air/internal_tank
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/_mecha.dm:81]: ?

File: code/modules/vehicles/mecha/_mecha.dm [hunk 2]
```diff
@@ -170,20 +171,32 @@
 	hud_possible = list (DIAG_STAT_HUD, DIAG_BATT_HUD, DIAG_MECH_HUD, DIAG_TRACK_HUD)
 
 /obj/item/radio/mech //this has to go somewhere
+	subspace_transmission = TRUE
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/_mecha.dm:174]: ?

File: code/modules/vehicles/mecha/_mecha.dm [hunk 3]
```diff
@@ -547,33 +538,29 @@
 		return
 	if(internal_damage & MECHA_INT_CONTROL_LOST)
 		target = pick(view(3,target))
-		if(!target)
-			return
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/_mecha.dm:551]: ?

File: code/modules/vehicles/mecha/_mecha.dm [hunk 4]
```diff
@@ -547,33 +538,29 @@
...
 		if((selected.range & MECHA_MELEE) && Adjacent(target))
-			if(isliving(target) && selected.harmful && HAS_TRAIT(L, TRAIT_PACIFISM))
-				to_chat(L, "<span class='warning'>You don't want to harm other living beings!</span>")
+			if(isliving(target) && selected.harmful && HAS_TRAIT(livinguser, TRAIT_PACIFISM))
+				to_chat(livinguser, "<span class='warning'>You don't want to harm other living beings!</span>")
 				return
 			INVOKE_ASYNC(selected, /obj/item/mecha_parts/mecha_equipment.proc/action, user, target, params)
 			return
-	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_MECHA_MELEE_ATTACK) || !istype(target, /atom) || !Adjacent(target))
+	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_MECHA_MELEE_ATTACK) || !Adjacent(target))
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/_mecha.dm:555]: Why's this a timer cooldown? I know it's not a part of the pr, just curious.

File: code/modules/vehicles/mecha/_mecha.dm [hunk 5]
```diff
@@ -547,33 +538,29 @@
...
 		return
 	if(internal_damage & MECHA_INT_CONTROL_LOST)
-		var/list/possible_targets = oview(1,src)
-		if(!length(possible_targets))
-			return
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/_mecha.dm:571]: ?

File: code/modules/vehicles/mecha/_mecha.dm [hunk 6]
```diff
@@ -692,26 +679,19 @@
...
-		var/area/destination_area = destination.loc
-		if(destination_area.area_flags & NOTELEPORT)
-			to_chat(occupants, "[icon2html(src, occupants)]<span class='warning'>A dull, universal force is preventing you from phasing here!</span>")
-			spark_system.start()
-		else
-			forceMove(destination)//This is jank I hate it thanks, this should be done thrugh move not this dumb shit
-		use_power(phasing_energy_drain)
+	. = ..()
+	if(phasing) //Theres only one cause for phasing canpass fails
 		addtimer(VARSET_CALLBACK(src, movedelay, TRUE), movedelay*3)
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/_mecha.dm:691]: Should this be here still?

File: code/modules/vehicles/mecha/_mecha.dm [hunk 7]
```diff
@@ -1082,32 +1033,34 @@
...
 			AI.remote_control = null
 			mob_container = AI
 			newloc = get_turf(AI.linked_core)
 			qdel(AI.linked_core)
 	else
 		return ..()
-	var/mob/living/L = M
+	var/mob/living/ejector = M
 	mecha_flags  &= ~SILICON_PILOT
-	if(mob_container.forceMove(newloc))//ejecting mob container
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/_mecha.dm:1098]: ?

File: code/modules/vehicles/mecha/_mecha.dm [hunk 8]
```diff
@@ -1174,6 +1127,37 @@
...
+		return cabin_air
+	return ..()
+
+/obj/vehicle/sealed/mecha/return_analyzable_air()
+	return cabin_air
+
+///fetches pressure of the gas mixture we are using
+/obj/vehicle/sealed/mecha/proc/return_pressure()
+	var/datum/gas_mixture/t_air = return_air()
+	return t_air?.return_pressure()
```
Comment (LemonInTheDark) [inline code/modules/vehicles/mecha/_mecha.dm:1152]: Just call it air, you're not "guaranteed" it's turf air, and this var name is ugly

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/mecha/_mecha.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:81]: Internal tank is always an air tank

File: code/modules/vehicles/mecha/_mecha.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:174]: Please clarify what you mean here

File: code/modules/vehicles/mecha/_mecha.dm (hunk 3 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:551]: Please clarify what you mean here

File: code/modules/vehicles/mecha/_mecha.dm (hunk 5 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:571]: Please clarify what you mean here

File: code/modules/vehicles/mecha/_mecha.dm (hunk 6 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:691]: Yes

File: code/modules/vehicles/mecha/_mecha.dm (hunk 7 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:1098]: Please clarify what you mean here

File: code/modules/vehicles/mecha/_mecha.dm (hunk 8 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:1152]: Change done

File: code/modules/vehicles/mecha/_mecha.dm (hunk 4 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:555]: it's a pretty uncommon cooldown + considering /mecha already has 3 million vars on it

File: code/modules/vehicles/mecha/_mecha.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/vehicles/mecha/_mecha.dm:174]: Why set subspace_transmission to TRUE? am I missing something?

File: code/modules/vehicles/mecha/_mecha.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/modules/vehicles/mecha/_mecha.dm:551]: I was confused about why the sanity check was removed. On closer inspection it was shitcode

File: code/modules/vehicles/mecha/_mecha.dm (hunk 5 above)
Comment (LemonInTheDark) [reply code/modules/vehicles/mecha/_mecha.dm:571]: See above

File: code/modules/vehicles/mecha/_mecha.dm (hunk 7 above)
Comment (LemonInTheDark) [reply code/modules/vehicles/mecha/_mecha.dm:1098]: Why remove the check here? Was it redundant?

Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:1098]: Correct

File: code/modules/vehicles/mecha/_mecha.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/vehicles/mecha/_mecha.dm:174]: It was manually set previously when the radio was added

**Pull request merged**